compiler-core/src/format.rs

```diff
@@ -150,21 +157,24 @@ impl<'comments> Formatter<'comments> {
 
     pub(crate) fn module<'a>(&mut self, module: &'a UntypedModule) -> Document<'a> {
         let mut documents = vec![];
-        let mut previous_was_import = false;
-
-        for definition in &module.definitions {
-            let is_import = definition.definition.is_import();
 
-            if documents.is_empty() {
-                // We don't insert empty lines before the first definition
-            } else if previous_was_import && is_import {
-                documents.push(lines(1));
+        // Here we take consecutive groups of imports so that they can be sorted
+        // alphabetically.
+        for (is_import_group, definitions) in &module
+            .definitions
+            .iter()
+            .group_by(|definition| definition.definition.is_import())
```

Comment:
Use `partition` or `partition_in_place` rather than `group_by` in situations like these please

Reply:
Wouldn't that hold a different output though? Using partition would yield two collections: one of imports and one with non imports, that means that the formatter would move all imports to the top of the document:
```gleam
import wibble
pub fn wobble()
import wobble
// Turns into:
import wibble
import wobble
pub fn wobble()
```
I did that because the current formatter's behaviour is to keep imports where those are defined and not move them at the top of the file. What should I do?

Reply:
ohhhh it produces multiple groups! Sorry this is my mistake, I thought it made `HashMap<bool, Definition>`. OK cool this is great